pennylane/devices/qubit/simulate.py

```diff
@@ -51,6 +51,7 @@ def simulate(circuit: qml.tape.QuantumScript) -> tuple:
     state = create_initial_state(circuit.wires, circuit._prep[0] if circuit._prep else None)
 
     for op in circuit._ops:
-        state = apply_operation(op, state)
+        batched_state = qml.math.ndim(state) != len(circuit.wires)
```

Comment:
this doesn't change in the loop, please move it up one line

Reply:
It can change in the loop. `batch_dim` is essentially a boolean checking whether the state is batched or not. Unless the first operation being applied to the state is broadcasted, the state will have no batch dimension until a broadcasted operation gets applied to it. But once the state is broadcasted, then its batch dimension will not change.

Reply:
Another way we could do this is by initializing it to `0` and then setting it to `1` encoutering an operator with a batch dimension.  
Basically 
```
batch_state = 1 if op.batch_size is not None else batch_state
```

Reply:
That would check if the operation is batched or not. We need to check if the state is batched or not.

Reply:
The state will have batching only after we have encountered an operation that has batching.

So we can initialize the `batched_state` to zero or the state prep batching. Then we set it to `1` after we applied an operation with a batch size.

Reply:
right.. it does in fact change mid-application. I think I just ask that you add a comment saying this can change mid-loop

Reply:
Just realized we will need to integrate batching with measurements before `simulate` will work with broadcasted parameters.  

I suggest we break changing `measure` task into a separate PR, and hold off on changing `simulate` until we have the measure function working with a batched state